### PR TITLE
feat(ui5-menu):  refactor menu items display

### PR DESCRIPTION
**Pull request by unazko**
- The `ui5-menu-item` now extends `ui5-li-custom` class and it
will be represented directly as a list item in the DOM.
As a result the` ui5-menu-item` is no longer an abstract class,
but a physical component and it will ease the possibility
in the future to support custom content supplied by
the application developers.

- The `ui5-reponsive-popover` is no longer
rendered into the static area. Thus the `ui5-menu` structure
defined by the application developers is represented more
accurately into the DOM and the browser `popover` API could
be used once available for all supported browsers.

- There is now no differentiation between mobile and desktop
device in regards to the display mechanism. In both cases
we rely on the template to do the job as the components used
for composition like `ui5-list` and `ui5-responsive-popover` do
comply with the device.

- There is also a new behavior implemented, which enables
the end users to navigate via mouse and keyboard to a
disabled `ui5-menu-item` element. This behavior is aimed
to improve the accessibility of the `ui5-menu` component
by providing screen reader announcements for disabled
menu items as well.

- The `haspopup` attribute is now applied to the `ui5-menu item`'s
with a sub-menu.

- The `ui5-menu` elements used for sub-menus are created only once
and are being reused afterwards. They are no longer destroyed on close.
This contributes to lowering the count of the slow DOM manipulation operations.

- The custom focus handling is now done solely in the `ui5-menu` component
as we no longer use the `ui5-responsive-popover` functionalities to retrieve the focus.
As a result the code is easier to maintain and trace the focus shifting.

Fixes: #7096
Fixes: #7767
Fixes: #7423
Related to: #6350
Related to: #8214
Related to: #7391

### CODE REVIEW

File: packages/main/src/MenuItem.ts [hunk 1]
```diff
@@ -15,22 +20,25 @@ import type Menu from "./Menu.js";
  *
  * <h3>Usage</h3>
  *
- * <code>ui5-menu-item</code> is an abstract element, representing a node in a <code>ui5-menu</code>. The menu itself is rendered as a list,
- * and each <code>ui5-menu-item</code> is represented by a list item (<code>ui5-li</code>) in that list. Therefore, you should only use
+ * <code>ui5-menu-item</code> is representing a node in a <code>ui5-menu</code>. The menu itself is rendered as a list,
```
Comment (BorisDafov) [inline packages/main/src/MenuItem.ts:23]: <code>ui5-menu-item</code> represents ...

File: packages/main/src/NavigationMenuItem.ts [hunk 2]
```diff
@@ -12,8 +12,8 @@ import MenuItem from "./MenuItem.js";
  *
  * <h3>Usage</h3>
  *
- * <code>ui5-navigation-menu-item</code> is an abstract element, representing a node in a <code>ui5-navigation-menu</code>. The navigation menu itself is rendered as a list,
- * and each <code>ui5-navigation-menu-item</code> is represented by a list item (<code>ui5-li</code>) in that list. Therefore, you should only use
+ * <code>ui5-navigation-menu-item</code> is representing a node in a <code>ui5-navigation-menu</code>. The navigation menu itself is rendered as a list,
```
Comment (BorisDafov) [inline packages/main/src/NavigationMenuItem.ts:15]: <code>ui5-navigation-menu-item</code represents ...

**Review by BorisDafov — CHANGES_REQUESTED** [review]
minor

File: packages/main/src/MenuItem.ts (hunk 1 above)
Comment (unazko) [reply packages/main/src/MenuItem.ts]: done

File: packages/main/src/NavigationMenuItem.ts (hunk 2 above)
Comment (unazko) [reply packages/main/src/NavigationMenuItem.ts]: done

File: packages/main/src/List.ts [hunk 3]
```diff
@@ -1040,7 +1040,7 @@ class List extends UI5Element {
 
 	focusFirstSelectedItem() {
 		// only enabled items are focusable
-		const firstSelectedItem = this.getFirstItem(x => x.selected && !x.disabled);
+		const firstSelectedItem = this.getFirstItem(x => x.selected && !x._focusable);
```
Comment (tsanislavgatev) [inline packages/main/src/List.ts:1043]: Should this be kept as not focusable? Might be just a missed conversion to -> x._focusable

File: packages/main/src/Menu.hbs [hunk 4]
```diff
@@ -55,54 +54,11 @@
...
-				{{else if this.item._siblingsWithChildren}}
-					<div
-						class="ui5-menu-item-no-icon-end"
-					>
-					</div>
-				{{/if}}
-				</div>
-			</ui5-li>
-		{{/each}}
+		<slot></slot>
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.hbs:61]: Do we need them as a slot or we need them as individual ones?

File: packages/main/src/Menu.ts [hunk 5]
```diff
@@ -416,18 +365,11 @@ class Menu extends UI5Element {
...
-			this._popover.resetFocus();
-		}
+		this._popover?.close(false, false, true);
 	}
 
-	async _createPopover() {
-		const staticAreaItemDomRef = await this.getStaticAreaItemDomRef();
-		this._popover = staticAreaItemDomRef!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
+	_createPopover() {
+		this._popover = this.shadowRoot!.querySelector<ResponsivePopover>("[ui5-responsive-popover]")!;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:372]: Should we always look for it or we can just keep the result  in the propety and look for it if it's empty?

File: packages/main/src/Menu.ts [hunk 6]
```diff
@@ -483,103 +417,88 @@ class Menu extends UI5Element {
...
 		}
 	}
 
-	_prepareSubMenuDesktopTablet(item: MenuItem, opener: HTMLElement, actionId: string) {
-		if (actionId !== this._subMenuOpenerId || (item && item.hasSubmenu)) {
-			// close opened sub-menu if there is any opened
-			this._closeItemSubMenu(this._openedSubMenuItem!, true);
-		}
+	async _prepareSubMenuDesktopTablet(item: MenuItem) {
+		// close opened sub-menu if there is any opened
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:458]: I think the name of the method _closeItemSubMenu is pretty self-explanatory and we don't need this comment

File: packages/main/src/Menu.ts [hunk 7]
```diff
@@ -606,36 +521,28 @@ class Menu extends UI5Element {
...
-
-			item.hasSubmenu && this._prepareSubMenuDesktopTablet(item, opener, hoverId);
-		} else if (isMenuClose && this._isSubMenu && this._parentMenuItem) {
-			const parentMenuItemParent = this._parentMenuItem.parentElement as Menu;
-			parentMenuItemParent._closeItemSubMenu(this._parentMenuItem, true);
+		if (shouldOpenMenu) {
+			const item = e.target as MenuItem;
+			item.hasSubmenu && await this._prepareSubMenuDesktopTablet(item);
+		} else if (shouldCloseMenu && this._isSubMenu && this._parentMenuItem) {
+			const parentItemMenu = this._parentMenuItem.parentElement as Menu;
```
Comment (tsanislavgatev) [inline packages/main/src/Menu.ts:535]: Can we have that interaction trough events? For example the menu to throw some event to the parent menu like "close-submenu" and then the parent to close it. I don't think that reaching to the parent and calling a method of the parent is good practice.

File: packages/main/src/themes/NavigationMenuItem.css [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+	min-height: var(--_ui5_side_navigation_item_height);
+	border-radius: 0.375rem;
+	transition: var(--_ui5_side_navigation_item_transition);
+	color: var(--sapList_TextColor);
+	font-weight: bold;
+	font-size: var(--sapFontSize);
+	border-bottom: none;
+}
+
+:host a {
```
Comment (tsanislavgatev) [inline packages/main/src/themes/NavigationMenuItem.css:27]: Do we have any other way to distinguish "a" elements or only by tagname?

File: packages/main/src/Menu.ts (hunk 5 above)
Comment (unazko) [reply packages/main/src/Menu.ts:372]: You're right. We don't have to perform redundant DOM operations here.
don

File: packages/main/src/themes/NavigationMenuItem.css (hunk 8 above)
Comment (unazko) [reply packages/main/src/themes/NavigationMenuItem.css:27]: done

File: packages/main/src/Menu.hbs (hunk 4 above)
Comment (unazko) [reply packages/main/src/Menu.hbs:61]: Actually it seems to me now that I've read the documentation that the individual slot scheme would be more suitable for the `ui5-navigation-menu` template as there is an anchor tag wrapper around most of the `ui5-navigation-menu-item`'s:

> Sometimes, however, each child must be placed separately in the shadow root, potentially wrapped in other HTML elements, to satisfy the UX design of the component.

The currently implemented slot scheme in the `ui5-menu` looks clearer. Not sure yet if we could use this slot approach in the `ui5-menu` and at the same time individual slot approach for the `ui5-navigation-menu` derived component.

**Comment by nnaydenow** [comment]
Hi @unazko, @tsanislavgatev,

Please play a bit with following scenario with the proposed implementation:
https://jsbin.com/cusozabusu/edit?html,output

**Review by vladitasev — CHANGES_REQUESTED** [review]
Let's not merge this PR until the popover API is supported for all browsers.

**Pull request closed without merging**